dashboard/api_test.py

```diff
@@ -490,42 +490,44 @@ def test_info_not_passed_offered(self, mock_format):
                 api.get_info_for_course(self.course, None)
             )
         # the mock object has been called 2 times
-        # one for the course that is current run
-        mock_format.assert_any_call(self.course_run, api.CourseStatus.OFFERED, None, position=1)
         # one for the one that is past
-        mock_format.assert_any_call(self.course_run_ver, api.CourseStatus.NOT_PASSED, None, position=2)
+        mock_format.assert_any_call(self.course_run_ver, api.CourseStatus.NOT_PASSED, None, position=1)
+        # one for the course that is current run
+        mock_format.assert_any_call(self.course_run, api.CourseStatus.OFFERED, None, position=2)
 
     @patch('dashboard.api.format_courserun_for_dashboard', autospec=True)
     def test_info_not_enrolled_not_passed_not_offered(self, mock_format):
         """test for get_info_for_course for course with run not passed and nothing offered"""
+        self.mmtrack.configure_mock(**{'has_passed_course.return_value': False})
         with patch(
             'dashboard.api.get_status_for_courserun',
             autospec=True,
             side_effect=self.get_mock_run_status_func(
-                api.CourseRunStatus.NOT_PASSED, self.course_run, api.CourseRunStatus.NOT_PASSED),
-        ), patch('courses.models.Course.get_next_run', autospec=True, return_value=None):
+                api.CourseRunStatus.CHECK_IF_PASSED, self.course_run, api.CourseRunStatus.CHECK_IF_PASSED),
+        ), patch('courses.models.Course.get_first_unexpired_run', autospec=True, return_value=None):
             self.assert_course_equal(
                 self.course,
-                api.get_info_for_course(self.course, None)
+                api.get_info_for_course(self.course, self.mmtrack)
             )
-        mock_format.assert_any_call(self.course_run, api.CourseStatus.NOT_PASSED, None, position=1)
-        mock_format.assert_any_call(self.course_run_ver, api.CourseStatus.NOT_PASSED, None, position=2)
+        mock_format.assert_any_call(self.course_run, api.CourseStatus.NOT_PASSED, self.mmtrack, position=1)
+        mock_format.assert_any_call(self.course_run_ver, api.CourseStatus.NOT_PASSED, self.mmtrack, position=2)
 
     @patch('dashboard.api.format_courserun_for_dashboard', autospec=True)
     def test_info_grade(self, mock_format):
         """test for get_info_for_course for course with a course current and another not passed"""
+        self.mmtrack.configure_mock(**{'has_passed_course.return_value': False})
         with patch(
             'dashboard.api.get_status_for_courserun',
             autospec=True,
             side_effect=self.get_mock_run_status_func(
-                api.CourseRunStatus.CURRENTLY_ENROLLED, self.course_run, api.CourseRunStatus.NOT_PASSED),
+                api.CourseRunStatus.CURRENTLY_ENROLLED, self.course_run, api.CourseRunStatus.CHECK_IF_PASSED),
```

Comment:
i remember now. for past course runs, we want to ignore any course runs that the user took as an audit (ie: they missed the deadline to upgrade). this if/else logic was being used to handle past course runs: `if (run_status.status == CourseRunStatus.CHECK_IF_PASSED and mmtrack.has_passed_course(run_status.course_run.edx_course_key)):`. that was made irrelevant with the choice to ignore `MISSED_DEADLINE` course runs. now essentially all we care about is non-audit course runs that have a grade, so that logic became an if/else with this condition: `run_status.status == CourseRunStatus.CHECK_IF_PASSED`. hence the change to this unit test